### PR TITLE
fix: tweak video-streams typings

### PR DESCRIPTION
**Pull request by JoVictorNunes**
### What does this PR do?

Fixes typyings of almost all webcam-related code except for the `video-provider` component. It depends on a lot of code which is not in typescript yet. Any attempt to fix it may cause problems. We must revisit it in the future when all webcam-related codebase is fully migrated to typescript.

This PR also implements some improvements for a few hooks.

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/connection-status/service.js [hunk 1]
```diff
@@ -204,8 +204,10 @@ const getAudioData = async () => {
  * @returns An Object containing video data for all video peers and screenshare
  *          peer
  */
-const getVideoData = async (streams) => {
-  const camerasData = await VideoService.getStats(streams) || {};
+const getVideoData = async (getVideoStreamsStats) => {
+  console.log(getVideoStreamsStats);
+  const camerasData = await getVideoStreamsStats() || {};
+  console.log(camerasData);
```
Comment (ramonlsouza) [inline bigbluebutton-html5/imports/ui/components/connection-status/service.js:210]:
```suggestion
```

File: bigbluebutton-html5/imports/ui/components/video-provider/video-provider-graphql/component.tsx [hunk 2]
```diff
@@ -411,33 +469,33 @@ class VideoProviderGraphql extends Component {
     }
   }
 
-  getStreamsToConnectAndDisconnect(streams) {
-    const streamsCameraIds = streams.filter(s => !s?.isGridItem).map(s => s.stream);
+  getStreamsToConnectAndDisconnect(streams: VideoItem[]) {
+    const streamsCameraIds = streams.filter((s) => s?.type !== 'grid').map((s) => (s as StreamItem).stream);
```
Comment (ramonlsouza) [inline bigbluebutton-html5/imports/ui/components/video-provider/video-provider-graphql/component.tsx:473]: is `type = 'grid'` replacing `isGridItem`? Do we still need both after the changes made in this PR?

Comment (JoVictorNunes) [reply bigbluebutton-html5/imports/ui/components/video-provider/video-provider-graphql/component.tsx:473]: > is type = 'grid' replacing isGridItem? 

Yes!

> Do we still need both after the changes made in this PR?

No, actually. Nice catch.

**Review by ramonlsouza — APPROVED** [review]

**Comment by antobinary** [comment]
:confetti_ball:

**Pull request merged**